cmd/api/http/handlers.go

```diff
@@ -75,12 +75,12 @@ func (h *BookHandler) archiveBook(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	responseJSON(w, http.StatusOK, archivedBook)
+	responseJSON(w, http.StatusOK, bookToResponse(archivedBook))
 }
 
 /* Validates the entry, then stores the entry as a new book. */
 func (h *BookHandler) createBook(w http.ResponseWriter, r *http.Request) {
-	var bookEntry book.Book
+	var bookEntry book.EntryBookRequest
```

Comment:
Have you tested this?
`book.EntryBookRequest` does not have the `json` tags, and I don't think it should. You can create new request types for the `http` layer here.

Reply:
Yes, I've tested with Postman yesterday and now again. Why should this fail?

Reply:
Thats because of the json Decoder? It accepts case-insensitive match, thats why it working, right?

> To unmarshal JSON into a struct, Unmarshal matches incoming object keys to the keys used by Marshal (either the struct field name or its tag), preferring an exact match but also accepting a case-insensitive match. By default, object keys which don't have a corresponding struct field are ignored (see Decoder.DisallowUnknownFields for an alternative).

But, if it's better, I can create an http layer request type exclusively for the decoding, and then copy its fields to the book layer struct, trouthg a function just like `bookToResponse`, but the opposite. You say...

Reply:
Its probably OK because of the go std lib default behavior, which is converting the field names to camel case when decoding a json object into a struct. Sometimes that is not the case. I would probably create the separate struct to be more consistent.